eth/downloader/downloader.go

```diff
@@ -664,7 +665,28 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 	}
 	p.log.Debug("Looking for common ancestor", "local", localHeight, "remote", remoteHeight)
 	if localHeight >= MaxForkAncestry {
+		// We're above the max reorg threshold, find the earliest fork point
 		floor = int64(localHeight - MaxForkAncestry)
+
+		// If we're doing a light sync, ensure the floor doesn't go below the CHT, as
+		// all headers before that point will be missing.
+		if d.mode == LightSync {
```

Comment:
Does this need to be conditional on LightSync? Other sync modes might set a non-zero genesis in the future.

Reply:
But other clients don't suffer from gaps in the ancestor list, right?

Reply:
Currently only light sync is broken. The reason I named it genesis was indeed so we may move the genesis block for full/fast sync too eventually, but that would be a lot more change and potentially breaking at it. I didn't want to screw around adding more stuff to a pre-release hotfix.

Reply:
And we currently don't support non-0 genesis blocks, so we can't even test it that it works. Better to be restrictive imho than to accidentally break something.